habitat/core/dataset.py

```diff
@@ -361,3 +373,47 @@ def _shuffle_iterator(self) -> None:
             episodes = list(self._iterator)
             random.shuffle(episodes)
             self._iterator = iter(episodes)
+
+    def step_taken(self):
+        self._step_count += 1
+
+    def _set_shuffle_intervals(self):
+        if self.max_scene_repetition_episodes > 0:
+            self._max_rep_episode = random.randint(
+                int(0.8 * self.max_scene_repetition_episodes),
```

Comment:
Let's create `repetition_rand_interval = 0.2` param for `EpisodeIterator` and do:
```
            self._max_rep_step = random.randint(
                int((1 - self.repetition_rand_interval) * self.max_scene_repetition_steps),
                int((1 + self.repetition_rand_interval) * self.max_scene_repetition_steps),
            )
```
Or even create util method:
```
@static
def _randomize_value(value, interval):
    return random.randint(
                int((1 - interval) * value),
                int((1 + interval) * value),
            )
```